builtins/generic.ispc

```diff
@@ -1863,8 +1891,33 @@ EXT PackedStoreResultType __packed_store_active2i64(uniform int8 *uniform ptr, v
     @llvm.masked.compressstore(val, ptr, mask);
     return __popcnt_int64(__cast_mask_to_i64(mask));
 }
-
+EXT PackedStoreResultType __packed_store_active2f16(uniform int8 *uniform ptr, varying float16 val, UIntMaskType mask) {
+    @llvm.masked.compressstore(val, ptr, mask);
+    return __popcnt_int64(__cast_mask_to_i64(mask));
+}
+EXT PackedStoreResultType __packed_store_active2f32(uniform int8 *uniform ptr, varying float val, UIntMaskType mask) {
+    @llvm.masked.compressstore(val, ptr, mask);
+    return __popcnt_int64(__cast_mask_to_i64(mask));
+}
+EXT PackedStoreResultType __packed_store_active2f64(uniform int8 *uniform ptr, varying double val, UIntMaskType mask) {
+    @llvm.masked.compressstore(val, ptr, mask);
+    return __popcnt_int64(__cast_mask_to_i64(mask));
+}
 // packed active loads
+EXT uniform int32 __packed_load_activei8(uniform int8 *uniform from, uniform int8 *uniform to, UIntMaskType mask) {
```

Comment:
With so many implementations, it would be better to define a template function `__packed_load_active` and call it in all `__packed_load_activeTYPE` functions. Unfortunately, we can't call intrinsics directly from template functions, so we need to create overloaded functions for `@llvm.masked.expandload`. See the change here: https://github.com/nurmukhametov/ispc/commit/43e4e2398e43b7321de5ecb84a4b76ea6dd74155. The same can be done with `__packed_store_active`.

Reply:
I agree that for `packed_load` it's beneficial - it removes code duplication. However for 2-lines code implementation for `packed_store` it seems as overkill. I tried it and `packed_store` implementation has optimized out https://github.com/aneshlya/ispc/commit/3b229495e5ec3b204a6983b789263f5d7225475f.

Reply:
Yes, for short functions, this might not make sense and could result in more code being written.